webdriver/tests/execute_async_script/pinning.py

```diff
@@ -0,0 +1,50 @@
+import pytest
+
+from webdriver.transport import Response
+
+from tests.support.asserts import assert_error, assert_success
+
+
+def pin_script(session, name, script):
+    body = {"script": script}
+
+    return session.transport.send(
+        "POST", "/session/{session_id}/execute/pin/{script_name}".format(
+            session_id=session.session_id,
+            script_name=name),
+        body)
+
+
+def execute_async_script(session, script, name=None, args=None):
+    if args is None:
+        args = []
+    body = {"script": script, "args": args}
+
+    return session.transport.send(
+        "POST", "/session/{session_id}/execute/async/{script_name}".format(
+            session_id=session.session_id,
+            script_name=name),
+        body)
+
+
+def test_execute_pinned(session):
+    pin_script(session, "One", "return 1;")
+    response = execute_async_script(session, name="One")
+    assert_success(response, 1)
+
+
+def test_name_and_script(session):
+    pin_script(session, "One", "return 1;")
+    response = execute_async_script(session, "return 1;", name="One")
+    assert_error(response, "invalid argument")
+
+
+def test_no_such_pinned_script(session):
+    response = execute_async_script(session, name="One")
+    assert_error(response, "no such script")
```

Comment:
You stored a script under this name in a different test, but we can’t be sure the state has been cleared. For this reason we need to restart the session between each of these tests. You can use the `new_session` fixture for this.

See some example usage here:
https://searchfox.org/mozilla-central/search?q=new_session&case=false&regexp=false&path=testing%2Fweb-platform%2Ftests%2Fwebdriver%2Ftests

Alternatively, it _may_ be possible to rely on the [proposed endpoint for clearing the pinned scripts](https://github.com/w3c/webdriver/pull/1445#discussion_r334833023), but this does create a little bit of inter-dependency between pinning and clearing scripts… I think overall that is better than restarting the session though, because session restarts are quite expensive.